File: src/main/java/ru/vood/enumFromTable/GenerateEnums.java
package ru.vood.enumFromTable;

import org.apache.maven.plugin.AbstractMojo;
import org.apache.maven.plugin.MojoExecutionException;
import org.apache.maven.plugin.MojoFailureException;
import org.apache.maven.plugins.annotations.LifecyclePhase;
import org.apache.maven.plugins.annotations.Mojo;
import org.apache.maven.plugins.annotations.Parameter;
import org.apache.maven.project.MavenProject;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.jdbc.datasource.DriverManagerDataSource;
import ru.vood.enumFromTable.jaxb2.PluginParamsXml;
import ru.vood.enumFromTable.run.MainRunnable;

import java.io.File;

@Mojo(name = "setname", defaultPhase = LifecyclePhase.GENERATE_SOURCES, threadSafe = true)
public class GenerateEnums extends AbstractMojo {

    @Parameter(property = "driverClassName")
    private String driverClassName;

    @Parameter(property = "username")
    private String username;

    @Parameter(property = "password")
    private String password;

    @Parameter(property = "url")
    private String url;

    @Parameter(property = "generatePackage")
    private String generatedPackage;

    @Parameter(property = "prefTable")
    private String prefTable;

    @Parameter(property = "propFile")
    private String propFile;

    @Parameter(defaultValue = "${project}", readonly = true)
    private MavenProject project;

    @Parameter
    private File configFile;

    private AnnotationConfigApplicationContext ctx;


    @Override
    public void execute() throws MojoExecutionException, MojoFailureException {

        PluginParamsXml PluginParamsXml;


        getLog().info("+++++++++++++++++GenerateEnums++++++++++++++++++++++");
        //ctx = new AnnotationConfigApplicationContext("ru.vood.enumFromTable");
        ctx = new AnnotationConfigApplicationContext(MainConfiguration.class);

        DriverManagerDataSource driverManagerDataSource = ctx.getBean("driverManagerDataSource", DriverManagerDataSource.class);

        driverManagerDataSource.setUrl(url);
        driverManagerDataSource.setDriverClassName(driverClassName);
        driverManagerDataSource.setUsername(username);
        driverManagerDataSource.setPassword(password);
        getLog().info("+++++++++++++++++driverManagerDataSource++++++++++++++++++++++");

        MainRunnable mainRunnable = ctx.getBean("mainRunnable", MainRunnable.class);
        System.out.println(mainRunnable);
        mainRunnable.run(prefTable);

/*
        getLog().info("+++++++++++++++++refresh++++++++++++++++++++++");
        ctx.refresh();
*/

    }
}

File: src/main/java/ru/vood/enumFromTable/GenerateEnums.kt
package ru.vood.enumFromTable

import javassist.ClassPool
import org.apache.maven.plugin.AbstractMojo
import org.apache.maven.plugin.MojoExecutionException
import org.apache.maven.plugins.annotations.LifecyclePhase
import org.apache.maven.plugins.annotations.Mojo
import org.apache.maven.plugins.annotations.Parameter
import org.apache.maven.project.MavenProject
import org.springframework.context.annotation.AnnotationConfigApplicationContext
import java.io.File


@Mojo(name = "setname", defaultPhase = LifecyclePhase.GENERATE_SOURCES, threadSafe = true)
open class GenerateEnums : AbstractMojo() {
    @Parameter(property = "generatePackage")
    private val generatedPackage: String? = null

    @Parameter(property = "propFile")
    private val propFile: String? = null

    @Parameter(defaultValue = "\${project}", readonly = true)
    private val project: MavenProject? = null

    @Parameter
    private val configFile: File? = null

    private var ctx: AnnotationConfigApplicationContext? = null

    @Throws(MojoExecutionException::class)
    override fun execute() {

        ctx = AnnotationConfigApplicationContext("ru.vood.enumFromTable")
        ctx!!.refresh()
/*
        ctx = AnnotationConfigApplicationContext(MainConfiguration::class.java)
        ctx!!.refresh()
*/


        //val annotationConfigApplicationContext = AnnotationConfigApplicationContext(GenerateEnums::class.java)
        //println("annotationConfigApplicationContext $annotationConfigApplicationContext")
        this.pluginContext.entries.forEach { println("getPluginContext -> $it") }
        this.

        log.info("=============================Generate Enum from table==================")
        log.info("Generate to package $generatedPackage")
        log.info("Generate to package $propFile")

        log.info("Configuration $configFile")

        project!!.runtimeClasspathElements.forEach { println("runtimeClasspathElements -> $it") }

        log.info("artifacts -> ")
        project.artifacts.forEach { log.info("artifacts -> $it") }

        log.info("stringPropertyNames -> ")
        project.properties.stringPropertyNames().forEach { log.info("stringPropertyNames -> $it") }

        val pool = ClassPool.getDefault()

        for (cp in project.compileClasspathElements) {
            log.info("compileClasspathElements -> $cp")
            //pool.insertClassPath(cp)
        }


/*
        val pool = ClassPool.getDefault()
        try {
            for (cp in project!!.compileClasspathElements) {
                pool.insertClassPath(cp)
            }
            val ct = pool.get("ru.easyjava.maven.Target")
            val field = ct.getField("NAME")
            log.info(String.format("%s %s", name, field.constantValue))
        } catch (e: NotFoundException) {
            log.error(e)
            throw MojoExecutionException(e.message)
        } catch (e: DependencyResolutionRequiredException) {
            log.error(e)
            throw MojoExecutionException(e.message)
        }
*/

    }


}